Composer/packages/lib/shared/src/copyUtils/shallowCopyAdaptiveAction.ts

```diff
@@ -0,0 +1,13 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { BaseSchema } from '../types';
+
+import { ExternalApi } from './ExternalApi';
+
+export function shallowCopyAdaptiveAction(input: BaseSchema, externalApi: ExternalApi): BaseSchema {
```

Comment:
```suggestion
export function shallowCopyAdaptiveAction<T = MicrosoftIDialog>(input: T, externalApi: ExternalApi): T {
```

Reply:
Template here is great! I will merge all following change requests to one commit instead of accept each of them

Reply:
We can either write a generic function here
```ts
function shallowCopy<T extends BaseSchema>(input: T): T {}
```
or use the `MicrosoftIDialog` type directly
```ts
function shallowCopy(input: MicrosoftIDialog): MicrosoftIDialog {}
```
`T = MicrosoftIDialog` seems not working since `MicrosoftIDialog` is already a set of types.
Now I use the second solution to avoid exporting `BaseSchema`.

Reply:
`MicrosoftIDialog` is a union of types, we cannot use it in a generic function.
![image](https://user-images.githubusercontent.com/8528761/68571843-ed409d00-049e-11ea-9515-f634633b35b7.png)

Reply:
There is another thing confused me. Semantically, schema ought have a `MicrosoftIAction` interface. In current botbuilder-dotnet schema, `actions` is not differentiated and typed with `MicrosoftIDialog` which means that it could be a recognizer / dialog / trigger. How do you understand that @a-b-r-o-w-n ?

Reply:
I like `function shallowCopy<T extends BaseSchema = BaseSchema>(input: T): T {}`

Reply:
Okay great!

Reply:
resolved